Dockerfile

```diff
@@ -25,32 +25,50 @@
 #     -i -t --rm \
 #     --runtime=nvidia \
 #     core3d/danesfield \
-#     danesfield/tools/material_classifier.py --cuda ...
+#     danesfield/tools/material_classifier.py --cuda ...    
+
+# Download and install miniconda3
+# Based on https://github.com/ContinuumIO/docker-images/blob/fd4cd9b/miniconda3/Dockerfile
+# The step to update 'conda' is necessary to avoid the following error when
+# downloading packages (see https://github.com/conda/conda/issues/6811):
+#
+#     IsADirectoryError(21, 'Is a directory')
+#
+
+FROM nvidia/cuda:10.0-devel-ubuntu18.04
 
-FROM nvidia/cuda:9.0-runtime-ubuntu16.04
 LABEL maintainer="Max Smolens <max.smolens@kitware.com>"
 
+COPY ply2txt.py ply2txt.py
+
+COPY patches /patches
```

Comment:
I see that the content of `patches` is used in subsequent steps, but if ply2txt is not used, you can move it to the end of the Dockerfile.  I tend to put these at the end if possible because if you change ply2txt.py it invalidates the cache for the later layers when building the container (even if it's not really a dependency for the next steps).